src/main/java/dev/ecattez/shahmat/board/pawn/EnPassantRankVisitor.java

```diff
@@ -0,0 +1,17 @@
+package dev.ecattez.shahmat.board.pawn;
+
+import dev.ecattez.shahmat.board.PieceColorVisitor;
+
+public class EnPassantRankVisitor implements PieceColorVisitor<Integer> {
+
+    @Override
+    public Integer visitBlack() {
+        return 4;
```

Comment:
Those values feel like magic number to me. Wouldn't it be clearer to use a `Rank` enum ?